### ISSUE TITLE
EAS Build: iOS build fails with the message that "could not find compatible versions of ExpoModulesCore"

### ISSUE THREAD

**Issue by ankitmaloo** [expo/expo#15249]
### Summary

I was trying to build on EAS planning to release an iOS build. 

Did everything according to documentation. eas configure, and then ran eas build for ios. Build failed with following error: 

Command `pod install` failed.
└─ Cause: CocoaPods could not find compatible versions for pod "ExpoModulesCore":
  In Podfile:
    EXImagePicker (from `../node_modules/expo-image-picker/ios`) was resolved to 10.2.3, which depends on
      ExpoModulesCore

    ExpoModulesCore (from `../node_modules/expo-app-loading/node_modules/expo-modules-core/ios`)

Specs satisfying the `ExpoModulesCore (from `../node_modules/expo-app-loading/node_modules/expo-modules-core/ios`), ExpoModulesCore` dependency were found, but they required a higher minimum deployment target.

Logs from expo:

# Build output

```log
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/<username>/Desktop/test/static/rneasb/noomirnapp/ios/Noomi.xcodeproj -configuration Debug -scheme project -destination id=EF23C07B-AF76-40DE-AB34-7E95B00F8D4D

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

note: Using new build system
note: Building targets in parallel
note: Planning build
note: Analyzing workspace
note: Constructing build description
note: Build preparation complete
<<repo>>/ios/Pods/Target Support Files/Pods-/Pods-.debug.xcconfig:1:1: error: unable to open file (in target "project" in project "project")
<<repo>>/ios/Pods/Target Support Files/Pods-/Pods-.debug.xcconfig:1:1: error: unable to open file (in target "project" in project "project")
<<repo>>/ios/Pods/Target Support Files/Pods-/Pods-.debug.xcconfig:1:1: error: unable to open file (in target "project" in project "project")
<<repo>>/ios/Pods/Target Support Files/Pods-/Pods-.debug.xcconfig:1:1: error: unable to open file (in target "project" in project "project")
warning: Capabilities for Signing & Capabilities may not function correctly because its entitlements use a placeholder team ID. To resolve this, select a development team in the 


```

# Error output

```log
** BUILD FAILED **


### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

42

### Environment

EAS CLI 0.35.0 environment info:
    System:
      OS: macOS 11.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 16.1.0 - /usr/local/bin/node
      Yarn: 1.22.11 - /usr/local/bin/yarn
      npm: 7.17.0 - /usr/local/bin/npm
    Utilities:
      Git: 2.24.0 - /usr/local/bin/git
    npmPackages:
      expo: ^42.0.0 => 42.0.5 
      expo-dev-client: ^0.4.7 => 0.4.7 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: ~0.63.4 => 0.63.4 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      eas-cli: 0.35.0
      expo-cli: 4.13.0

### Reproducible demo or steps to reproduce from a blank project

1. Start with the basic project. 
2. Install expo-image-picker as evidently that is what is causing the issue. 
3. run expo configure
4. run expo run:ios

**Comment by filiptronicek** [comment]
I am experiencing a similar issue with my own project, but not sure if it is (even remotely) the same.

```
[!] Unable to find a specification for `UMCore` depended upon by `EXClipboard`
```

<details>
<summary>Further info and logs</summary>

In the `Install Pods` step of my EAS Build, it outputs this:
```
Installing pods
Using Expo modules
Auto-linking React Native modules for target `Interclip`: RNCClipboard, RNCMaskedView, RNGestureHandler, RNReanimated, RNSVG, RNScreens, react-native-appearance, react-native-camera, react-native-netinfo, and react-native-safe-area-context
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
Fetching podspec for `RCT-Folly` from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
Adding spec repo `trunk` with CDN `https://cdn.cocoapods.org/`
CocoaPods 1.11.2 is available.
To update use: `sudo gem install cocoapods`
For more information, see https://blog.cocoapods.org and the CHANGELOG for this version at https://github.com/CocoaPods/CocoaPods/releases/tag/1.11.2
[!] Unable to find a specification for `UMCore` depended upon by `EXClipboard`
You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
[stderr] [!] `<PBXResourcesBuildPhase UUID=`13B07F8E1A680F5B00A75B9A`>` attempted to initialize an object with an unknown UUID. `B534F3281A1C49B2B25725B3` for attribute: `files`. This can be the result of a merge and the unknown UUID is being discarded.
pod exited with non-zero code: 1
```

</details>

**Comment by ankitmaloo** [comment]
Since I posted this thread, i managed to get a workaround to make it work somehow. 

The only file you need to look at is package.json. Everything comes from there. 
Delete everything you don't absolutely need, bring it to a point where you know it will work with no surprises. 

Then keep adding the packages - 5 at a time, and keep running pod install. Skip the one which throws an error. 

Once you get all the problematic ones, look at their version numbers. Try with higher ones, lower ones, be careful about 
"~" or"^" - these fuck things up. I found my error in the expo-image-picker. I had 10.2.2 installed, and it needed 10.2.3. Put that in and i managed to get one successful build with what I need.

**Comment by brentvatne** [comment]
this can happen if there are multiple versions of expo-modules-core in your project or you're using a version of a library that is incompatible. in sdk 44 we'll move away from depending directly on expo-modules-core in expo sdk packages (https://github.com/expo/expo/pull/15205), but for now you can use yarn resolutions or whichever method you prefer to ensure that the expo-modules-core dep used by your expo sdk packages is the same. the reason this occurs is that if you install eg: expo-camera that depends on expo-modules-core ~0.4.0 and expo-clipboard that depends on, for example, expo-modules-core ~0.4.2, your package manager may not dedupe the dependency and resolve to the same version if they are installed at a different time.

**Comment by lc3t35** [comment]
It seems that EXLocalization has been forgotten by https://github.com/expo/expo/pull/13757 ?

android : 

```
expo prebuild -p android 
...
A problem occurred evaluating project ':expo-localization'.
> Project with path ':unimodules-core' could not be found in project ':expo-localization'.
```

ios : 

I've tried to fix changing UM by EX in .h and .m, so pod install works fine 

```
EXLocalization.h
// Copyright 2018-present 650 Industries. All rights reserved.

#import <ExpoModulesCore/EXExportedModule.h>

@interface EXLocalization : EXExportedModule
@end
```

```
#import <EXLocalization/EXLocalization.h>

@implementation EXLocalization

EX_EXPORT_MODULE(ExpoLocalization)
...
EX_EXPORT_METHOD_AS(getLocalizationAsync,
                    getLocalizationAsync:(EXPromiseResolveBlock)resolve
                    rejecter:(EXPromiseRejectBlock)reject)
{
  resolve([self constantsToExport]);
}
```

after expo run:ios I have this error

```
Unable to resolve module @unimodules/core from /xxxx/node_modules/expo-localization/build/ExpoLocalization.js: @unimodules/core could not be found within the project or in these directories:
  node_modules

import { NativeModulesProxy } from '@unimodules/core';                            ^
  2 | export default NativeModulesProxy.ExpoLocalization;
```

Can you please quickly update expo-localization package ?

**Comment by ricohumme** [comment]
Hi,

I'm having these issues as well. In my case the component giving me issues was  EXSplashScreen.
Going in to source where EXSplashScreen was located I checked the Podfile there and found the following:
`/node_modules/expo-app-loading/node_modules/expo-splash-screen/ios/EXSplashScreen.podspec`
```
...
  s.platform       = :ios, '12.0'
...
```
So I decided to checkout the project on a Mac and eject it from Expo and found the following in the Podfile used by the managed workflow when ejecting
```
...
  s.platform       = :ios, '11.0'
...
```
So I think there is a version mismatch resulting in the error you and I are facing.
My question here is: How can I set for which version I am building in a managed workflow?

**Comment by RupamShaw** [comment]
same  issue, Is there any solution instead of upgrading to sdk 42 . I am having sdk 41 and reactnative 63.4. This issue came up on installing expo-av it needs ios 12. so upgraded now can't revert back to 11 .

```

pod install` failed.
└─ Cause: CocoaPods could not find compatible versions for pod "EXSplashScreen":
  In Podfile:
    EXSplashScreen (from `../node_modules/expo-app-loading/node_modules/expo-splash-screen/ios`)

Specs satisfying the `EXSplashScreen (from `../node_modules/expo-app-loading/node_modules/expo-splash-screen/ios`)` dependency were found, but they required a higher minimum deployment target.

```

**Comment by jack2684** [comment]
Similar issue when trying to install mapbox

```
/Users/junjieguan/git/tickcell/react-native/ios/Pods/Target Support Files/Pods-TickCell/Pods-TickCell.debug.xcconfig:1:1: error: unable to open file (in target "TickCell" in project "TickCell")
/Users/junjieguan/git/tickcell/react-native/ios/Pods/Target Support Files/Pods-TickCell/Pods-TickCell.debug.xcconfig:1:1: error: unable to open file (in target "TickCell" in project "TickCell")
/Users/junjieguan/git/tickcell/react-native/ios/Pods/Target Support Files/Pods-TickCell/Pods-TickCell.debug.xcconfig:1:1: error: unable to open file (in target "TickCell" in project "TickCell")
/Users/junjieguan/git/tickcell/react-native/ios/Pods/Target Support Files/Pods-TickCell/Pods-TickCell.debug.xcconfig:1:1: error: unable to open file (in target "TickCell" in project "TickCell")
warning: Capabilities for Signing & Capabilities may not function correctly because its entitlements use a placeholder team ID. To resolve this, select a development team in the TickCell editor. (in target 'TickCell' from project 'TickCell')

```

https://github.com/react-native-mapbox-gl/maps/blob/master/plugin/install.md

**Comment by brentvatne** [comment]
i'm going to close this issue because there isn't a reproducible example available. if you are encountering this problem:

1) make sure you don't have any unimodules packages in your sdk 43+ project. this means no react-native-unimodules, @unimodules/core, or @unimodules/react-native-adapter.
2) ensure that you only have one version of expo-modules-core (eg: `yarn why expo-modules-core`). you *do not* need to install this package directly. in sdk 44 it is a dependency of the expo package in order to prevent duplicate versions.

if you run into a related problem, please open a new issue and share a reproducible example. thank you!

**Comment by yunying1** [comment]
I have same question, because dependency with expo not compatible, so you need use `expo install` reinstall all dependency.
I hope this will help you.
